src/libcore/iter/iterator.rs

```diff
@@ -258,6 +258,32 @@ pub trait Iterator {
         None
     }
 
+    /// Returns an iterator adaptor that goes over the original iterator,
+    /// skipping `step` elements at a time.
+    ///
+    /// **Panics** if `step == 0`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let a = [0, 1, 2, 3, 4, 5];
+    ///
+    /// let mut iter = a.iter().every_nth(2);
+    ///
+    /// assert_eq!(iter.next(), Some(&0));
+    /// assert_eq!(iter.next(), Some(&2));
+    /// assert_eq!(iter.next(), Some(&4));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    #[inline]
+    #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
Because if we rename `every_nth` to `step_by`, then there's going to be a method name conflict for ranges, because there would be a `step_by` from the Iterator trait, and a [`step_by` from the Range struct](https://doc.rust-lang.org/std/ops/struct.Range.html#method.step_by).

Technically we wouldn't need to remove the `Step` trait, but if we switch the `Range::step_by` implementation to the iterator version, then there's going to be no use for the `Step` trait in the core lib. Besides, it looks like the only implementors of `Step` are the normal integers. Nobody really uses the `Step` trait to increment an integer when they can just use `+= 1`. And if people really wanted it, we could just implement `Iterator` for them and have them return the integer plus one. It would work the same.